science/imager/Portfile

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           compilers 1.0
+PortGroup           active_variants 1.1
+
+name                imager
+version             202211
+set my_version      [string tolower [clock format [clock scan 2000-[string range ${version} 4 5]-10] -format %b]][string range ${version} 2 3][string range ${version} 6 end]
+supported_archs     arm64 x86_64
+set my_machine      [if {[string match *arm* ${os.arch}]} {list arm64} {list x86_64}]
+categories          science
+platforms           darwin
+maintainers         {u-bordeaux.fr:julien.penguen @penguen}
+license             permissive
+
+description         Radio-interferometry data analysis software
+
+long_description    IMAGER (and its associated visualization tool VIEWER) is a state-of-the-art \
+                    software oriented toward (sub-)millimeter radioastronomical interferometry. \ 
+                    It is used to reduce data acquired with the IRAM NOEMA interferometer \
+                    and the ALMA or JVLA arrays, with a simple, convenient user interface.
+
+homepage            https://imager.oasu.u-bordeaux.fr
+master_sites        https://nuage.oasu.u-bordeaux.fr/index.php/s/xEfBsde3zyTgo6x \
+                    https://nuage.oasu.u-bordeaux.fr/index.php/s/BKtnCwwj6nz76NG
+distname            ${name}-${my_version}
+use_xz              yes
+
+checksums           rmd160  9c1185a5c5e9fc54612808977ee8f548b2258d31 \
+                    sha256  c5d0288e3e95700655219d2617a3f109ffc2a37fbab983877c3c73410c7df952 \
+                    size    60725540
+
+patch.pre_args      -p1
+patchfiles          patch-admin-Makefile.def.diff \
+                    patch-admin-Makefile.python.diff \
+                    patch-admin-Makefile.build.diff \
+                    patch-admin-gildas-env.sh.diff \
+                    patch-admin-define-system.sh.diff
+
+depends_lib         path:lib/pkgconfig/gtk+-2.0.pc:gtk2 \
+                    port:libpng \
+                    port:python310 \
+                    port:py310-numpy
```

Comment:
As of January 1, 2023, the MacPorts recommended version of Python is 3.11. There are three lines in the Portfile where this currently needs to be changed. Consider defining a variable for this so that there is only a single line where it needs to be changed it in the future, since the recommended version of Python will increase every January 1.

Reply:
correction done